### DISCUSSION TITLE
Help me make my first Open source contribution

### DISCUSSION THREAD

**Discussion by tomkoevoets** [community/community#123108] [Programming Help]
### Hello github Community!

I'm excited to be reaching out to you today as I'm eager to contribute an open source project. However, I'm new to the world of open source and could use some guidance on where to begin.

I have a school assignment coming up where I'll be required to contribute to an open source project, making this opportunity perfect for me to learn and grow.
For this the school project my contribution has to be a small (good first issue) and a bigger one.

This far i have learned html, css, and java.
I would really love to do a project that contributes to something good in the world or something nature related.
In the end the most important thing is that i find a project with a quick response so i can start working.

Could someone please advise me on how I can get started? I'd greatly appreciate any help or guidance you can offer as I embark on this exciting journey into the world of open source.

Thank you in advance for your support!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by sajadh76** [comment]
Hi!

Take a look at [Finding ways to contribute to open source on GitHub]( https://docs.github.com/en/get-started/exploring-projects-on-github/finding-ways-to-contribute-to-open-source-on-github)

Also [First Contributions: Help beginners to contribute to open source projects](https://github.com/firstcontributions/first-contributions)

Good Luck :)

**Comment by MarcosApodaca** [comment]
# Step 1: Fork the project
This is as simple as clicking a button on GitHub. Navigate to the repository of the project you wish to contribute, then click on the fork button on the top left as seen in the image below.

After using the fork button you will have the repository in your GitHub account.

![Click here to fork](https://github.com/community/community/assets/112188399/7f41c591-f13f-4d33-9b98-0dc53c9c4f35)

# Step 2: Clone the project on your local machine
This is the simplest part of Git. Navigate to the forked repository (The repository is now in your project list on GitHub). After step 1 and 2 as you can see in the image below you must copy the url path to clone the repository. 

Then you must clone the project with the command 
``git clone <the copied url>``

![1](https://github.com/community/community/assets/112188399/0d95a52f-9a33-4544-a189-4c5b3d4f403f)

# Step 3: Create the branch where you are going to work
Creating a new branch in GitHub is a simple process. Here are the steps to follow:

Navigate to your repository on GitHub.

Click the drop-down button that says "branch: main" (or the name of the current branch) at the top of the page.
Type the name of your new branch in the text box. Branch names must be unique within a repository.
Press the Enter key and that's it! You have created a new branch.
If you prefer to do it from the command line, you can follow these steps:

``git branch (your_branch_name)``
Then switch to the newly created branch
``git checkout (your_branch_name)``

Now you can make the code change in that branch, without interfering with the main branch

# Step 4: Git add and commit your contribution

``git add``
``git commit -m 'Commit message'``

To push the changes you have been working on you should run:
``git push origin <branch-name>``

now you have the changes in your branch


# Step 5: Create a pull request
This is the final step for any contribution to an open project, you will be saying "I've made some changes, would you mind adding them to the project?

By opening a pull request, you will be hoping that the project owner or members like what they see and mix them in. Otherwise, they may make changes or request that you make them before they blend them into the stable branch of the project.

To open a pull request, navigate to the main repository as you can see in the following image. You will see the last branch you uploaded 'login-feature', then click on 'compare and pull request'.
![1 (2)](https://github.com/community/community/assets/112188399/d87cb853-418b-49f1-90b9-3b3cb2593275)

Explain clearly the changes you have made and then create the pull request as you can see in the image:

![1 (3)](https://github.com/community/community/assets/112188399/8fa35e5f-ee01-492c-9972-e0abcbcbcbce)

This is all, I hope it is helpful, if you want to know a little more you can check this Github link.
https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project
